include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _UATTRIBUTESBUILDER_
+#define _UATTRIBUTESBUILDER_
+
+#include <memory>
+#include <up-core-api/uri.pb.h>
+#include <up-cpp/uuid/factory/UuidFactory.h>
+#include <string>
+#include <spdlog/spdlog.h>
+#include "up-core-api/uattributes.pb.h"
+#include "up-core-api/uuid.pb.h"
+#include "up-core-api/umessage.pb.h"
+
+using namespace uprotocol::uri;
+using namespace uprotocol::v1;
+
+namespace uprotocol {
+
+class UAttributesBuilder {
+    private:
+        uprotocol::v1::UAttributes attributes_;
+
+    public:
+        UAttributesBuilder() {
```

Comment:
you need also the constructor for the mandatory attributes 
            UAttributes(const UUID id,
                        const UMessageType type,
                        const UPriority priority) {